src/main/java/com/sysdig/jenkins/plugins/sysdig/CLIDownloadAction.java

```diff
@@ -0,0 +1,260 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Scanner;
+
+import hudson.model.Run;
+import jenkins.model.RunAction2;
+
+public class CLIDownloadAction implements RunAction2 {
+    private String name;
+    private String os;
+    private String arch;
+    private String version;
+    private transient Run run;
+    private String url;
+    private String basePath = "/var/jenkins_home";
+
+    private Path apath;
+
+    public String chwd() {
+
+        return this.apath.toAbsolutePath().toString();
+    }
+
+    private void makeUrl() {
+        try {
+            @SuppressWarnings("deprecation")
+            URL uarl = new URL("https", "download.sysdig.com", "/scanning/sysdig-cli-scanner/latest_version.txt");
+            Scanner s = new Scanner(uarl.openStream());
+            if (this.version.isEmpty() || this.version.equalsIgnoreCase("latest")){
+                this.version = s.nextLine();
+            }
+            this.url = String.format(
+                    "https://download.sysdig.com/scanning/bin/sysdig-cli-scanner/%s/%s/%s/sysdig-cli-scanner",
+                    this.version,
+                    this.os.toLowerCase(), this.arch.toLowerCase());
+            s.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+     private void downloadUsingNIO( String file) throws IOException {
+        URL url = new URL(this.url);
+        ReadableByteChannel rbc = Channels.newChannel(url.openStream());
+        FileOutputStream fos = new FileOutputStream(file);
+        fos.getChannel().transferFrom(rbc, 0, Long.MAX_VALUE);
+        fos.close();
+        rbc.close();
+    }
+
+    private void downloaderStream(String p) throws Exception {
+        URL url = new URL(this.url);
+        BufferedInputStream bis = new BufferedInputStream(url.openStream());
+        FileOutputStream fis = new FileOutputStream(p);
+        byte[] buffer = new byte[1024];
+        int count = 0;
+        while ((count = bis.read(buffer, 0, 1024)) != -1) {
+            fis.write(buffer, 0, count);
+        }
+        fis.close();
+        bis.close();
+    }
+    
+    private void downloader() throws Exception {
+        if ("".equals(this.url)) {
+            throw new Exception("empty url");
+        }
+        String tmpPath = String.format("%s/sysdig-cli-scanner.%s", this.basePath, this.version);
+        try {
+            downloaderStream(tmpPath);
+        } catch (Exception e) {
+            
+            downloadUsingNIO(tmpPath);
+        }
+        
+    }
+
+    public CLIDownloadAction(String name, String basePath,String ver) throws Exception {
+        this.name = name;
+        this.basePath = basePath;
+        this.os = System.getProperty("os.name");
+        this.version = ver;
+        this.arch = System.getProperty("os.arch");
+        makeUrl();
+        downloader();
```

Comment:
the name of the action is CLIDownloader makeUrl (for cli downloader) but i renamed those i don't understand why you want these private methods to return a value instead of just putting the correct value where it should be if you have a specific reason pleased teoo